explorer/client/src/components/pagination/Pagination.tsx

```diff
@@ -151,7 +151,7 @@ function Pagination({
         label: string = String(finalPageNo)
     ) => (
         <button
-            id="lastBtn"
+            data-testid="lastBtn"
```

Comment:
adding test identifiers to the production code feels backwards.  why not have the tests use real ids ?

Reply:
IDs aren't really intended for this, I'd argue that we're using them wrong. They're also supposed to be globally unique, which we are violating in a handful of cases. Globally unique IDs also goes against reusable components. Most modern approaches to `id`'s actually generate them in a way that wouldn't be targetable (e.g. React's new `useId()` hook). Targeting without using test IDs depends on either using DOM structure or other not-super-stable properties. Using test ID's like this is the current best practice in the community, and has been adopted most places that do end-to-end testing. It gives a lot of stability to tests.

For these ending up in production code that's fair, the argument generally goes:
- Production code often includes mechanisms to make testing easier.
- These are small enough, so it won't really matter for performance / bundle size / etc.
- There are plugins in the ecosystem that exist to strip these identifiers from production. I'm actually not convinced this is ideal, because it prevents you from running E2E tests in actual production environments, which is neat to be able to do.

Reply:
ok, makes sense re: reusable components, what if you have more than one on screen etc. 
 i do think stripping the test ids out would be good but that's fine for later

Reply:
Resolving this since we also discussed on slack.